python/tank/bootstrap/resolver.py

```diff
@@ -295,8 +296,60 @@ def _get_pipeline_configurations_for_project(self, pipeline_config_name, current
         )
 
         for pc in pipeline_configs:
+            # We'll need to provide a descriptor object for the config if
+            # possible. Note that it's possible that we'll be returning a
+            # None for the config descriptor. It's up to other filtering
+            # operations to remove those if desired.
+            #
+            # As in resolve_shotgun_configuration, the order of precedence
+            # is as follows:
+            #
+            # 1. windows/linux/mac path
+            # 2. descriptor
+            # 3. sg_descriptor
             path = ShotgunPath.from_shotgun_dict(pc)
 
+            try:
+                current_os_path = path.current_os
+            except ValueError:
+                current_os_path = None
+
+            uri = pc.get("descriptor") or pc.get("sg_descriptor")
+
+            if path:
+                # Make sure that the config has a path for the current OS.
+                if current_os_path is None:
+                    log.debug("Config isn't setup for %s: %s", sys.path, pc)
+                    cfg_descriptor = None
+                else:
+                    cfg_descriptor = create_descriptor(
+                        sg_connection,
+                        Descriptor.CONFIG,
+                        dict(path=current_os_path, type="path"),
+                    )
+            elif uri:
+                log.debug("Using descriptor uri: %s", uri)
+                cfg_descriptor = create_descriptor(
+                    sg_connection,
+                    Descriptor.CONFIG,
+                    uri,
+                )
+            else:
+                # If we have neither a uri, nor a path, then we can't get
+                # a descriptor for this config.
+                log.debug("No uri or path found for config: %s", pc)
+                cfg_descriptor = None
+
+            # We add to the pc dict even if the descriptor is a None. It'll be
+            # the responsibility of the filter call below to remove any pcs
+            # that are deemed to be not useful.
+            if cfg_descriptor is None:
+                log.debug("Unable to create descriptor for config: %s", pc)
+            else:
+                log.debug("Config descriptor created: %r", cfg_descriptor)
+
+            pc["config_descriptor"] = cfg_descriptor
+
```

Comment:
I'm looking at the comments a couple of lines below this one and it reminded me that we are returning pipeline configurations without the current os being set on purpose, and it all has to do with the Shotgun Desktop.

The way we know today (in the real world, not in zero config) if we can run the Pipeline configuration wizard is if there is no pipeline for a project. Therefore, the code in the manager will always return pipeline configurations who have at least one platform configured, even if the current one isn't. This allows us to know if Desktop if we're in an error state or not.

For plugin_id-based pipelines we've decided that it wouldn't work this way and that you needed all the relevant fields to be set or the pipeline configuration would be skipped.

The fact that no unit tests have discovered this bug you've introduced indicates I have forgotten to add such a test in.

As such, having one path but not the current_os path set shouldn't be considered as a reason to skip a pipeline configuration. You'll probably want to have an empty descriptor then. Which means the websocket code can't assume to always get pipelines with a descriptor.

Sorry to have you go back and forth on this.